### ISSUE TITLE
CMake error:  "X11::xcb"  not found

### ISSUE THREAD

**Issue by dongdongbh** [jstkdng/ueberzugpp#7]
```
Target "ueberzug" links to target "X11::xcb" but the target was not found.
Perhaps a find_package() call is missing for an IMPORTED target, or an
ALIAS target is missing?
```
`libx11-xcb-dev` already installed.
Test system: `Ubuntu 20.04 server`

**Comment by jstkdng** [comment]
Can you show the complete cmake output?

**Comment by dongdongbh** [comment]
The full cmake output:
```bash
$ build cmake -DCMAKE_BUILD_TYPE=Release ..
-- The CXX compiler identification is GNU 9.4.0
-- Check for working CXX compiler: /usr/bin/c++
-- Check for working CXX compiler: /usr/bin/c++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Found nlohmann_json: /usr/lib/cmake/nlohmann_json/nlohmann_jsonConfig.cmake (found version "3.7.3")
-- Found X11: /usr/include
-- Looking for XOpenDisplay in /usr/lib/x86_64-linux-gnu/libX11.so;/usr/lib/x86_64-linux-gnu/libXext.so
-- Looking for XOpenDisplay in /usr/lib/x86_64-linux-gnu/libX11.so;/usr/lib/x86_64-linux-gnu/libXext.so - found
-- Looking for gethostbyname
-- Looking for gethostbyname - found
-- Looking for connect
-- Looking for connect - found
-- Looking for remove
-- Looking for remove - found
-- Looking for shmat
-- Looking for shmat - found
-- Looking for IceConnectionNumber in ICE
-- Looking for IceConnectionNumber in ICE - found
-- Found PkgConfig: /usr/bin/pkg-config (found version "0.29.1")
-- Looking for C++ include pthread.h
-- Looking for C++ include pthread.h - found
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD - Failed
-- Looking for pthread_create in pthreads
-- Looking for pthread_create in pthreads - not found
-- Looking for pthread_create in pthread
-- Looking for pthread_create in pthread - found
-- Found Threads: TRUE
-- Found OpenCV: /usr (found version "4.2.0") found components: core imgproc imgcodecs videoio
-- Checking for module 'xcb-image'
--   Found xcb-image, version 0.4.0
-- Checking for module 'vips-cpp'
--   Found vips-cpp, version 8.9.1
-- Checking for module 'libsixel'
--   Found libsixel, version 1.8.2
-- Configuring done
CMake Error at CMakeLists.txt:57 (add_executable):
  Target "ueberzug" links to target "X11::xcb" but the target was not found.
  Perhaps a find_package() call is missing for an IMPORTED target, or an
  ALIAS target is missing?


-- Generating done
CMake Generate step failed.  Build files cannot be regenerated correctly.
```

**Comment by jstkdng** [comment]
which cmake version are you using? seems the X11::xcb target was added since 3.18.

**Comment by dongdongbh** [comment]
I'm using 3.16, so I have to upgrade to 3.18 to make this work?

**Comment by jstkdng** [comment]
ok, could be an issue with your compiler. Are you able to upgrade to gcc 12?

**Comment by dongdongbh** [comment]
I upgraded gcc to 11.1, but it still shows same error.
```bash
$ gcc --version
gcc (Ubuntu 11.1.0-1ubuntu1~20.04) 11.1.0
Copyright (C) 2021 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
```

**Comment by jstkdng** [comment]
same issue with jthread? can you delete the build file and run cmake again?

**Comment by dongdongbh** [comment]
Yes, same error, maybe it dues to I forgot to link g++ to new version (I only link gcc), I will link g++ and test it next week, and try again.

In your updated readme, you said `cmake <= 3.18`, I guess it should be `>=`.

**Comment by jstkdng** [comment]
>In your updated readme, you said cmake <= 3.18, I guess it should be >=.

Ty for catching that, I'll fix that.

So, did you manage to fix the issue?

**Comment by dongdongbh** [comment]
Hi, I manage to fix it by upgrading g++ to v10.3, so both gcc and g++ should be updated.
